### PR TITLE
Identity refactor

### PR DESCRIPTION
**Pull request by nfx**

### CODE REVIEW

**Comment by nfx** [comment]
@stikkireddy consider this as almost finished one. Still need to add few tests and perhaps `databricks_user_instance_profile` resource

File: identity/resource_user.go [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+		if err != nil {
+			return diag.FromErr(err)
+		}
+		err = internal.StructToData(entity, s, d)
+		if err != nil {
+			return diag.FromErr(err)
+		}
+		return nil
+	}
+}
```
Comment (stikkireddy) [inline identity/resource_user.go:30]: Hmm seems a bit obscure to make this a public method and only use it in the readContext. The method definition is also quite confusing to read in the readContext.

maybe something like:
```go
entity = NewUsersAPI(m).ReadR(d.Id())
return Bind(entity)(d, userSchema)
```

**Review by stikkireddy — CHANGES_REQUESTED** [review]

File: identity/resource_user.go (hunk 1 above)
Comment (nfx) [reply identity/resource_user.go]: removed this method

**Review by nfx — COMMENTED** [review]
applied some review comments

**Review by stikkireddy — APPROVED** [review]
LGTM, need to run the integration tests for the release.

**Pull request merged**